src/Microsoft.Diagnostics.ExtensionCommands/ParallelStacks.Runtime/StackFrame.cs

```diff
@@ -13,22 +14,21 @@ public class StackFrame
     {
         public string TypeName { get; private set; }
         public string MethodName { get; private set; }
-        public List<string> Signature { get; }
-
-        public string Text { get; }
+        public string Text { get; private set; }
+        public List<string> Signature { get; } = new List<string>();
 
         public StackFrame(ClrStackFrame frame)
         {
-            var signature = frame.Method?.Signature;
-            Text = string.IsNullOrEmpty(signature) ? "?" : string.Intern(signature);
-            Signature = new List<string>();
+            if (frame.Method is null)
```

Comment:
I would avoid throwing an exception (especially in a constructor).
Instead, compute the different properties based on the fact that no method description is available

Reply:
This actually works as "should not be reached" assert. StackFrames are not supposed to be created with null-methods. (at least from what I see in code). I can remove this check but I'm not sure what kind of default values should we assign? Same as when trypeName/methodName/signature are nulls?

Reply:
I've fixed this anyway and removed the check. It's just that I'm personally a null-hater and I'd rahter change ctor to receive not a stackFrame but an actual non-null Method instance and introduced StackFrame.Empty property with default values so the caller code can explicitly use it. List would be IReadOnlyList then to make this type readonly-like.